core/src/test/scala/weaponregex/parser/ParserJSTest.scala

```diff
@@ -0,0 +1,895 @@
+package weaponregex.parser
+
+import scala.util.Failure
+import weaponregex.model.regextree._
+
+class ParserJSTest extends munit.FunSuite {
```

Comment:
Perhaps all the common tests for both flavors can be extracted into a common trait test suite, with added tests for each flavor in the implementation? The two parser tests have many duplicate tests like this

Reply:
This makes sense. We will definitely look into this 👍